app/controllers/analysis_controller.rb

```diff
@@ -0,0 +1,420 @@
+    elsif decedent_race.length > 0 && decedent_age_range.length > 0
+      common_races.each do |race|
+        modified_key = race + " " + partial_age_hash.select{|a| a === age_at_death}.values.first
+        if decedent_frequency[modified_key].key?(record_death_year)
+          decedent_frequency[modified_key][record_death_year] += 1
+        else
+          decedent_frequency[modified_key][record_death_year] = 1
+        end
+      end
+    # Place into the hash based on the key containing the sex and age range of the decedent.
+    elsif decedent_sex.length > 0 && decedent_age_range.length > 0
+      modified_key = record_sex + " " + partial_age_hash.select{|a| a === age_at_death}.values.first
+      if decedent_frequency[modified_key].key?(record_death_year)
+        decedent_frequency[modified_key][record_death_year] += 1
+      else
+        decedent_frequency[modified_key][record_death_year] = 1
+      end
+    # Place into the hash based on the key containing just the races of decedents.
+    elsif decedent_race.length > 0
+      common_races.each do |race|
+        if decedent_frequency[race].key?(record_death_year)
+          decedent_frequency[race][record_death_year] += 1
+        else
+          decedent_frequency[race][record_death_year] = 1
+        end
+      end
+    # Place into the hash based on the key containing just the sex of the decedent.
+    elsif decedent_sex.length > 0
+      if decedent_frequency[record_sex].key?(record_death_year)
+        decedent_frequency[record_sex][record_death_year] += 1
+      else
+        decedent_frequency[record_sex][record_death_year] = 1
+      end
+    # Place into the hash based on the key containing just the age range.
+    elsif decedent_age_range.length > 0
+      if decedent_frequency[partial_age_hash.select{|a| a === age_at_death}.values.first].key?(record_death_year)
+        decedent_frequency[partial_age_hash.select{|a| a === age_at_death}.values.first][record_death_year] += 1
+      else
+        decedent_frequency[partial_age_hash.select{|a| a === age_at_death}.values.first][record_death_year] = 1
+      end
+    else
+      if decedent_frequency.key?("All Individuals")
+        if decedent_frequency["All Individuals"].key?(record_death_year)
+          decedent_frequency["All Individuals"][record_death_year] += 1
+        else
+          decedent_frequency["All Individuals"][record_death_year] = 1
+        end
+      end
+    end
+  end
+  # For all of the years in each of the decedent type hashes that had no deaths in a particular year, make the value 0.
+  (start_year..end_year).each do |year|
+    decedent_frequency.each_value do |value|
+      if !value.key?(year)
+        value[year] = 0
+      end
+    end
+  end
+  render json: decedent_frequency
+end
+
+  # Create a whitelist of params; these are generated dynamically from the
+  # curently active step. This means the only parameters accepted at any
+  # point is scoped by the active Step.
```

Comment:
It looks like this comment isn't accurate for this use case. I think you might have taken it from another part of the application?